### PR TITLE
ci: fix saucelabs browser crashing due to idle timeout

### PR DESCRIPTION
**Pull request by devversion**
See commit message

### CODE REVIEW

**Comment by itea-dev** [comment]
Nit: hopeing => hoping

**Comment by devversion** [comment]
Fixing the timeout/disconnect and updating to iOS14 reduced the 30min time to 10min

File: tools/package-tools/ts-compile.ts [hunk 1]
```diff
@@ -11,7 +11,9 @@ import * as chalk from 'chalk';
 export function tsCompile(binary: 'tsc' | 'ngc', flags: string[]) {
   return new Promise<void>((resolve, reject) => {
     const binaryPath = resolvePath(`./node_modules/typescript/bin/${binary}`);
-    const childProcess = spawn(binaryPath, flags, {shell: true});
+    // Use `node` for spawning the TypeScript compiler. On Windows, the shell is
+    // not necessarily able to detect that the `tsc` files relies on Node.
+    const childProcess = spawn('node', [binaryPath, ...flags], {shell: true});
```
Comment (crisbeto) [inline tools/package-tools/ts-compile.ts:16]: I ran into this when debugging something else locally through Gulp. I think it has to do with the fact that we install multiple TS versions for the 4.2 compatibility check and we end up picking out the older version's binary.

**Review by crisbeto — APPROVED** [review]
LGTM

File: tools/package-tools/ts-compile.ts (hunk 1 above)
Comment (devversion) [reply tools/package-tools/ts-compile.ts:16]: Yeah, there are actually two issues. The first is the one you mentioned. i.e. `node_modules/.bin/tsc` is a different version than the one we installed explicitly, and the second one is (as per my comment) that `node_modules/typescript/bin/tsc` is a NodeJS script but on Windows the shebang is not interpreted properly.

I assume we could add another comment but this the legacy setup anyway 😄

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request merged**